### PR TITLE
remove IronPython code and extension

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

This PR removes DSIronPython, IronPythonExtension, IronPythonTests and as much of the old completion provider code as possible as they are replaced by:

https://github.com/DynamoDS/DSIronPython

It also remove obsolete events.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### CODE REVIEW

File: src/Libraries/PythonNodeModelsWpf/IronPythonCompletionData.cs [hunk 1]
```diff
@@ -89,31 +69,17 @@ public System.Windows.Media.ImageSource Image
...
-        private string _description;
-
-        public object Description
-        {
-            get
-            {
-                // lazily get the description
-                if (_description == null)
-                {
-                    _description = provider.GetDescription(this.Stub, this.Text, this.IsInstance).TrimEnd('\r', '\n');
```
Comment (aparajit-pratap) [inline src/Libraries/PythonNodeModelsWpf/IronPythonCompletionData.cs:105]: Is it okay to get rid of this logic `provider.GetDescription(this.Stub, this.Text, this.IsInstance).TrimEnd('\r', '\n')`?

File: src/Libraries/PythonNodeModelsWpf/IronPythonCompletionData.cs [hunk 2]
```diff
@@ -15,7 +15,6 @@ public class IronPythonCompletionData : ICompletionData
     {
```
Comment (aparajit-pratap) [inline src/Libraries/PythonNodeModelsWpf/IronPythonCompletionData.cs:15]: Is this class also used for autocomplete in PythonNet?

Comment (mjkkirschner) [reply src/Libraries/PythonNodeModelsWpf/IronPythonCompletionData.cs:15]: no, it's used to decouple AvalonEdit's interface (and WPF deps) from the core completion types. We can think of this as a UI type, thats why it's in this PythonWPF assembly.

File: src/Libraries/PythonNodeModelsWpf/IronPythonCompletionData.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/Libraries/PythonNodeModelsWpf/IronPythonCompletionData.cs:105]: yes, it's safe, this type should have as little logic as possible. This logic already exists in the `PythonCodeCompletionDataCore.Description` property that we use to set this objects description in the constructor.

`PythonCodeCompletionDataCore` is an `IExternalCompletionData` object which the constructor on line 50 now takes, and that is the only constructor for this object.

**Review by aparajit-pratap — APPROVED** [review]

**Pull request merged**